memgpt/agent.py

```diff
@@ -568,6 +568,17 @@ def step(self, user_message, first_message=False, first_message_retry_limit=FIRS
             if user_message is not None:
                 self.interface.user_message(user_message)
                 packed_user_message = {"role": "user", "content": user_message}
+                # Special handling for AutoGen messages with 'name' field
+                try:
+                    user_message_json = json.loads(user_message)
+                    # Treat 'name' as a special field

```

Comment:
@sarahwooders and FYI for the storage refactor - AutoGen seems to be using `name` to mark different users:

<img width="1005" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/54a6ee21-81ab-4f08-9a0e-b55c91768a7d">

We should probably do the same or at least respect usage like this for integrations like AutoGen (which is part of this PR). For the storage refactor, we should probably treat `name` as the same as `content` and `role` and store them all at the same level in the schema.

Reply:
More background here: https://github.com/microsoft/autogen/issues/268